lib/game.rb

```diff
@@ -23,14 +23,30 @@ def make_move
     @display.display_board(@board.board)
     @display.show_current_player(@current_player.mark)
     move = @display.ask_for_move
-    @board.player_make_move(@current_player.mark, move)
-    @display.display_board(@board.board)
+    until move_valid?(move)
+      @display.invalid_move_message(move)
+      move = @display.ask_for_move
+    end
+    @board.player_make_move(@current_player.mark, move.to_i)
   end
 
   public
 
+  def moves_remaining?
```

Comment:
moves_remaining? and moves_remaining is a little confusing... Could this be re-phrased as:

```rb
def complete?
  @board.moves_remaining == 0
end
```

Reply:
Thanks, Matt